Docs 7754 dwformats duke (#1319)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-formats-java.adoc

=== Example: Use a Generic

This example relies on generic support in the class name to create a `java.util.ArrayList` of `User` objects.

Note that you do not need to use a generic to specify the class in each instance. The class is taken from the generic in the list.

[source,dataweave,linenums]
----
output application/json
---
[{
    name: "Tomo",
    lastName: "Chibana"
},
{
    name: "Ana",
    lastName: "Felissati"
},
{
    name: "Leandro",
    lastName: "Shokida"
}
] as Array {class: "java.util.ArrayList<User>"}
----

// CONFIG PROPS ///////////////////////////////////////////////////////

[[properties]]
== Configuration Properties

DataWeave supports the following configuration properties for the Java format.

=== Reader Properties

There are no reader properties for Java data.

=== Writer Properties

The Java reader accepts properties that provide instructions for writing output data.

[cols="1,1,1,3a", options="header"]
|===
| Parameter | Type | Default | Description
| `duplicateKeyAsArray` | `Boolean` | `false` | If duplicate keys are detected in
an object, the writer will change the value to an array with all those values.
Valid values are `true` or `false`.

> | `writeAttributes` | `Boolean` | `false` | If a key has attributes, it will put

Author: ```suggestion
| `writeAttributes` | `Boolean` | `false` | Indicates whether to add attributes of a key to children of the key. The new attribute key name will start with &amp;. 
```

> | `writeAttributes` | `Boolean` | `false` | If a key has attributes, it will put

Author: ```suggestion
| `writeAttributes` | `Boolean` | `false` | Indicates whether to add attributes of a key to children of the key. The new attribute key name will start with &#64;. Valid options are `true` or `false`.
```

> them as children key-value pairs of the key that contains them. The attribute

Author: ```suggestion
```

> them as children key-value pairs of the key that contains them. The attribute

Author: ```suggestion
```

> key name will start with @. Valid values are `true` or `false`.

Author: ```suggestion
Valid values are `true` or `false`.
```

> key name will start with @. Valid values are `true` or `false`.

Author: ```suggestion
```
